floors/rule_test.go

```diff
@@ -854,3 +850,352 @@ func TestGenerateCombinations(t *testing.T) {
 		})
 	}
 }
+
+func TestGetDeviceType(t *testing.T) {
+	tests := []struct {
+		name    string
+		request *openrtb2.BidRequest
+		want    string
+	}{
+		{
+			name:    "user agent contains device type as tablet",
+			request: &openrtb2.BidRequest{Device: &openrtb2.Device{UA: "Mozilla/5.0 (Windows NT touch 10.0; Win64; x64)"}},
+			want:    "tablet",
+		},
+		{
+			name:    "user agent contains Android.*Mobile",
+			request: &openrtb2.BidRequest{Device: &openrtb2.Device{UA: "Mozilla/5.0 (Android Redmi Mobile; Win64; x64)"}},
+			want:    "phone",
+		},
```

Comment:
Do you want to add one more test case where `UA` contains `Phone`?

Reply:
added more UT's for `phone`